lib/seattleflu/id3c/cli/command/etl/redcap_det_scan.py

```diff
@@ -170,6 +170,96 @@ def locations(db: DatabaseSession, cache: TTLCache, record: dict) -> list:
     return [tract_entry, address_entry]
 
 
+def zipcode_map(redcap_code: str) -> str:
+    """
+    Maps *redcap_code* to corresponding zip code. This is required because the
+    branching logic in REDCap cauases major issues if the code is the
+    actual zip code.
+    """
+    zipcode_map = {
+        '44': '98101',
+        '45': '98102',
+        '46': '98103',
+        '47': '98104',
+        '48': '98105',
+        '49': '98106',
+        '50': '98107',
+        '51': '98108',
+        '52': '98109',
+        '53': '98112',
+        '54': '98115',
+        '55': '98116',
+        '56': '98117',
+        '57': '98118',
+        '58': '98119',
+        '59': '98121',
+        '60': '98122',
+        '61': '98125',
+        '62': '98126',
+        '63': '98133',
+        '64': '98134',
+        '65': '98136',
+        '66': '98144',
+        '67': '98146',
+        '69': '98154',
+        '70': '98155',
+        '71': '98164',
+        '75': '98177',
+        '76': '98178',
+        '78': '98195',
+        '80': '98199',
+        '4':  '98004',
+        '5':  '98005',
+        '6':  '98006',
+        '7':  '98007',
+        '8':  '98008',
+        '21': '98033',
+        '22': '98034',
+        '25': '98039',
+        '26': '98040',
+        '33': '98056',
+        '36': '98059',
+        '10': '98011',
+        '17': '98028',
+        '1':  '98001',
+        '2':  '98002',
+        '3':  '98003',
+        '15': '98023',
+        '18': '98030',
+        '19': '98031',
+        '20': '98032',
+        '28': '98042',
+        '31': '98047',
+        '32': '98055',
+        '34': '98057',
+        '35': '98058',
+        '68': '98148',
+        '72': '98166',
+        '73': '98168',
+        '77': '98188',
+        '79': '98198',
+        '81': '98010',
+        '82': '98022',
+        '83': '98038',
+        '84': '98045',
+        '85': '98051',
+        '86': '98065',
+        '87': '98027',
+        '88': '98029',
+        '89': '98052',
+        '90': '98074',
+        '91': '98075',
+        '92': '98092',
+        '93': '98070'
```

Comment:
I think it's good practice to sort this sort of static list of data.

Sorting also reveals that there are gaps, which seems somewhat suspicious to me. The missing numbers are:

9
11..14
16
23..24
27
29..30
37..43
74

Reply:
I copied this over in the exact order from the SCAN REDCap codebook. Those values just seem to not be used as codes 🤷

Reply:
Alright! :-)